nion/swift/Inspector.py

```diff
@@ -918,245 +926,246 @@ def _set_property_value(self, property_value: typing.Any) -> None:
+        super().__init__()
+        self.document_controller = document_controller
+        self.display_item = display_item
+        self.display_layer = display_layer
+        self.label_model = DisplayLayerPropertyCommandModel(document_controller, display_item, display_layer, "label")
+
+        index = display_item.display_layers.index(display_layer)
+        display_data_channel = display_item.get_display_layer_display_data_channel(index)
+        data_index = display_item.display_data_channels.index(display_data_channel) if display_data_channel else None
+        self.data_index_model = Model.PropertyModel(data_index)
+
+        self.data_row_model = DisplayLayerPropertyCommandModel(document_controller, display_item, display_layer, "data_row")
+        self.fill_color_model = DisplayLayerPropertyCommandModel(document_controller, display_item, display_layer, "fill_color")
+        self.stroke_color_model = DisplayLayerPropertyCommandModel(document_controller, display_item, display_layer, "stroke_color")
+        self.stroke_width_model = DisplayLayerPropertyCommandModel(document_controller, display_item, display_layer, "stroke_width")
+
+        self.show_display_type_chooser = (self.display_layer.display_data_channel
+                                          and self.display_layer.display_data_channel.data_item
+                                          and self.display_layer.display_data_channel.data_item.is_data_complex_type)
+        self.display_type_items = [_("Log Absolute"), _("Absolute"), _("Phase"), _("Real"), _("Imaginary")]
+        self._display_type_flags = ["log-absolute", "absolute", "phase", "real", "imaginary"]
+        self._display_type_reverse_map = {p: i for i, p in enumerate(self._display_type_flags)}
+
+        current_flag = display_data_channel.complex_display_type if display_data_channel and display_data_channel.complex_display_type else str()
+        self.current_display_type_index_model = Model.PropertyModel[int](self._display_type_reverse_map.get(current_flag, 0))
+
+        self.__data_index_listener = self.data_index_model.property_changed_event.listen(
+            ReferenceCounting.weak_partial(LinePlotDisplayLayerModel.__on_data_index_changed, self))
+        self.__display_type_index_listener_model = self.current_display_type_index_model.property_changed_event.listen(
+            ReferenceCounting.weak_partial(LinePlotDisplayLayerModel.__on_display_type_index_changed, self))
+        self.__display_layer_list_added_listener = display_layers_list_model.item_added_event.listen(
+            ReferenceCounting.weak_partial(LinePlotDisplayLayerModel.__on_display_layer_list_changed, self))
+        self.__display_layer_list_removed_listener = display_layers_list_model.item_removed_event.listen(
+            ReferenceCounting.weak_partial(LinePlotDisplayLayerModel.__on_display_layer_list_changed, self))
```

Comment:
I'm not sure this is the best way to do things. I suspect this leads to the errors you mention below (old LinePlotDisplayLayerModel in list). This might take some more rework before we can handle the index properly. I think that we're going to have to refactor `ChangeDisplayLayerDisplayDataChannelCommand` to take a `DisplayLayer` directly and track it via uuid rather than index. That would be a separate PR that goes in _before_ this PR. Doing that would eliminate most uses of `index` except for the display in the text item - but that could be accomplished using the straightforward implementation of `index` (`return self.display_item.display_layers.index(self.display_layer)`).

Reply:
Ok I have a plan - let's eliminate the 'index' property from `LinePlotDisplayLayerModel`. To do this, `ChangeDisplayLayerDisplayDataChannelCommand` the move methods in `LinePlotDisplayLayerHandler` can calculate their indexes on the fly. That leaves only the static text item - and let's just not put the index there for now. That eliminates the `index` property and issue for now.